pkg/kubelet/volumes.go

```diff
@@ -77,12 +77,18 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.BoundPod) (volumeMap, error) {
 	for i := range pod.Spec.Volumes {
 		volSpec := &pod.Spec.Volumes[i]
 
+		podRef, err := api.GetReference(pod)
```

Comment:
@lavalamp that seems sensible to me

On Mon, Mar 9, 2015 at 5:19 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/kubelet/volumes.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5122#discussion_r26077446>
> :
>
> > @@ -77,12 +77,18 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.BoundPod) (volumeMap, error) {
> >  	for i := range pod.Spec.Volumes {
> >  		volSpec := &pod.Spec.Volumes[i]
> >
> > +		podRef, err := api.GetReference(pod)
>
> @markturansky <https://github.com/markturansky> It's not an etcd thing.
> It's that the object didn't come from apiserver, which fills in the self
> links.
>
> @pmorie <https://github.com/pmorie> IMO I think for resiliency we should
> make GetReference fail gracefully (return a mostly-complete
> ObjectReference) if it can't parse the self link.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5122/files#r26077446>
> .
>